### PR TITLE
Add linter scripts and check in workflow

### PR DESCRIPTION
**Pull request by pzread**
Copy related liner scripts from https://github.com/openxla/iree and run them in the workflow.

Several changes are made in linters to fit them into the repo. Changes can be found in the second commit: e9a1017313da130372897f7a50b3f8f108ee91b6

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Copyright 2023 The OpenXLA Authors
```
Comment (pzread) [inline .github/workflows/lint.yml:1]: I'm not sure if we need to relicense these files?

Comment (GMNGeoffrey) [reply .github/workflows/lint.yml:1]: This isn't relicensing. Just changing the authors line and this is a superset of the "IREE Authors" although I'm not actually sure that we've got it properly defined somewhere. We should probably put an AUTHORS file in the openxla/community repo, but this is fine for now

File: build_tools/scripts/run_markdownlint.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# https://github.com/igorshubovych/markdownlint-cli
+
+set -euo pipefail
+
+declare -a included_files_patterns=(
+  # All .md files (disabled while we decide how rigorously to apply lint checks)
+  # "./**/*.md"
+
+  # Just .md files for the user-facing website.
+  "./docs/website/**/*.md"
```
Comment (GMNGeoffrey) [inline build_tools/scripts/run_markdownlint.sh:19]: Looks not applicable

File: .github/workflows/lint.yml (hunk 1 above)
Comment (pzread) [reply .github/workflows/lint.yml:1]: Thanks for confirmation.

**Review by GMNGeoffrey — APPROVED** [review]

**Pull request merged**